web-app/HeimScripts/heatmap/run.R

```diff
@@ -189,118 +256,140 @@ formatSubset <- function(subsetNumber){
 ### duplicated code from utils - when we get sourcing to work it will be moved
 
 
-mergeFetchedData <- function(listOfHdd){
+mergeFetchedData <- function(listOfHdd) {
   df <- listOfHdd[[1]]
   expected.rowlen <- nrow(df)
   labels <- names(listOfHdd)
   df <- add.subset.label(df,labels[1])
-  if(length(listOfHdd) > 1){
-    for(i in 2:length(listOfHdd)){
+  if (length(listOfHdd) > 1) {
+    for (i in 2:length(listOfHdd)) {
       df2 <- listOfHdd[[i]]
       label <- labels[i]
       df2 <- add.subset.label(df2,label)
-      df <- merge(df, df2 ,by=c("Row.Label","Bio.marker"))
-      if(nrow(df) != expected.rowlen){
-        assign("errors", "Mismatched probe_ids - different platform used?", envir = .GlobalEnv)
+      df <- merge(df, df2 ,by = c("Row.Label","Bio.marker"))
+      if (nrow(df) != expected.rowlen) {
+        stop("Mismatched probe_ids - different platform used?")
```

Comment:
because the two approaches are not much different, and both give the correct result. 
one could even argu that mine is faster - as retrieving nrow twice and comparing it is faster that comparing n number of columns.